mesh/mesh.go

```diff
@@ -754,11 +758,14 @@ func (msh *Mesh) accumulateRewards(l *types.Layer, params Config) {
 			msh.With().Warning("Atx from block not found in db", log.Err(err), bl.ID(), bl.ATXID)
 			continue
 		}
-		ids = append(ids, atx.Coinbase)
-
+		coinbases = append(coinbases, atx.Coinbase)
+		//Sid: add a mapping between accounts and nodeID, pass this into the function
+		if _, exists := smeshers[atx.Coinbase]; !exists {
```

Comment:
Unfortunately this won't work since one coinbase may be associated with multiple smeshers (although the _reverse_ won't happen in a single layer). Use parallel arrays, or a struct, or some other solution to store a smesherID/nodeID for each block/coinbase address.

Reply:
Resolving this through our discussion